proxy/http/server/web.py

```diff
@@ -271,17 +281,19 @@ def _try_route(self, path: bytes) -> bool:
                 self.route = self.routes[protocol][route]
                 assert self.route
                 # Optionally, upgrade protocol
-                if do_ws_upgrade:
+                if do_ws_upgrade and self.route.do_upgrade(self.request):
                     self.switch_to_websocket()
                     assert self.route
                     # Invoke plugin.on_websocket_open
                     self.route.on_websocket_open()
                 else:
                     # Invoke plugin.handle_request
                     self.route.handle_request(self.request)
-                    if self.request.has_header(b'connection') and \
-                            self.request.header(b'connection').lower() == b'close':
-                        return True
+                    # if self.request.has_header(b'connection') and \
+                    #         self.request.header(b'connection').lower() == b'close':
+                    #     return True
```

Comment:
This was likely a bug which would result in closing the connection even before full response has been dispatched